### PR TITLE
Lot of tiny perfs

### PR DESCRIPTION
**Pull request by panquez**

### CODE REVIEW

File: src/geode/mesh/core/solid_mesh.cpp [hunk 1]
```diff
@@ -1230,6 +1230,8 @@ namespace geode
                     polyhedron_adj, f };
             }
         }
+        // SDEBUG( polyhedron_barycenter( polyhedron_adj ) );
+        // SDEBUG( polyhedron_barycenter( polyhedron_facet.polyhedron_id ) );
```
Comment (BotellaA) [inline src/geode/mesh/core/solid_mesh.cpp:1234]:
```suggestion
```

File: src/geode/mesh/core/tetrahedral_solid.cpp [hunk 2]
```diff
@@ -325,6 +322,85 @@ namespace geode
...
+            //                                     { { polyhedron_adj, f }, v }
+            //                                     ) )
+            //         == vertices.end() )
+            //     {
+            //         all_contained = false;
+            //         break;
+            //     }
+            // }
+            // if( all_contained )
+            // {
```
Comment (BotellaA) [inline src/geode/mesh/core/tetrahedral_solid.cpp:391]:
```suggestion
```

File: src/geode/mesh/core/tetrahedral_solid.cpp [hunk 3]
```diff
@@ -325,6 +322,85 @@ namespace geode
...
+            return absl::optional< PolyhedronFacet >{ absl::in_place,
+                polyhedron_adj, f };
+            // }
```
Comment (BotellaA) [inline src/geode/mesh/core/tetrahedral_solid.cpp:394]:
```suggestion
```

File: src/geode/mesh/core/tetrahedral_solid.cpp [hunk 4]
```diff
@@ -325,6 +322,85 @@ namespace geode
...
+        }
+        // SDEBUG( polyhedron_barycenter( polyhedron_adj ) );
+        // SDEBUG( polyhedron_barycenter( polyhedron_facet.polyhedron_id ) );
```
Comment (BotellaA) [inline src/geode/mesh/core/tetrahedral_solid.cpp:397]:
```suggestion
```

File: src/geode/mesh/core/tetrahedral_solid.cpp [hunk 5]
```diff
@@ -325,6 +322,85 @@ namespace geode
...
+            return absl::nullopt;
+        }
+        std::array< index_t, 3 > vertices;
+        for( const auto v : LRange{ 3 } )
+        {
+            vertices[v] =
+                this->polyhedron_facet_vertex( { polyhedron_facet, v } );
+        }
+        const auto polyhedron_adj = opt_polyhedron_adj.value();
+        const std::array< index_t, 4 > adj_vertices{
```
Comment (BotellaA) [inline src/geode/mesh/core/tetrahedral_solid.cpp:343]: Why not calling polyhedron_vertices()

File: src/geode/mesh/core/tetrahedral_solid.cpp [hunk 6]
```diff
@@ -325,6 +322,85 @@ namespace geode
...
+        TetrahedralSolid< dimension >::polyhedron_adjacent_facet(
+            const PolyhedronFacet& polyhedron_facet ) const
+    {
+        const auto opt_polyhedron_adj =
+            this->polyhedron_adjacent( polyhedron_facet );
+        if( !opt_polyhedron_adj )
+        {
+            return absl::nullopt;
+        }
+        std::array< index_t, 3 > vertices;
```
Comment (BotellaA) [inline src/geode/mesh/core/tetrahedral_solid.cpp:336]: Why not calling polyhedron_facet_vertices()

File: src/geode/mesh/core/tetrahedral_solid.cpp [hunk 7]
```diff
@@ -325,6 +322,85 @@ namespace geode
...
+                this->polyhedron_facet_vertex( { polyhedron_facet, v } );
+        }
+        const auto polyhedron_adj = opt_polyhedron_adj.value();
+        const std::array< index_t, 4 > adj_vertices{
+            this->polyhedron_vertex( { polyhedron_adj, 0 } ),
+            this->polyhedron_vertex( { polyhedron_adj, 1 } ),
+            this->polyhedron_vertex( { polyhedron_adj, 2 } ),
+            this->polyhedron_vertex( { polyhedron_adj, 3 } )
+        };
+        std::array< bool, 4 > candidates{ true, true, true, true };
```
Comment (BotellaA) [inline src/geode/mesh/core/tetrahedral_solid.cpp:349]:
```suggestion
        std::array< bool, 4 > candidates{ false, false, false, false };
```

File: src/geode/mesh/core/tetrahedral_solid.cpp [hunk 8]
```diff
@@ -325,6 +322,85 @@ namespace geode
...
+                && opp_adj_vertex_id != vertices[2] )
+            {
+                for( const auto other_v : LRange{ 4 } )
+                {
+                    if( other_v == v )
+                    {
+                        continue;
+                    }
+                    candidates[other_v] = false;
+                }
```
Comment (BotellaA) [inline src/geode/mesh/core/tetrahedral_solid.cpp:364]: candidates[v] = true;
break;

Comment (BotellaA) [reply src/geode/mesh/core/tetrahedral_solid.cpp:364]: maybe do not break in case of invalid mesh?

File: src/geode/mesh/core/tetrahedral_solid.cpp (hunk 5 above)
Comment (panquez) [reply src/geode/mesh/core/tetrahedral_solid.cpp:343]: It leads to more Ir

File: src/geode/mesh/core/tetrahedral_solid.cpp (hunk 6 above)
Comment (panquez) [reply src/geode/mesh/core/tetrahedral_solid.cpp:336]: Same

File: src/geode/mesh/core/tetrahedral_solid.cpp (hunk 5 above)
Comment (BotellaA) [reply src/geode/mesh/core/tetrahedral_solid.cpp:343]: Yes, probably but you might want to benchmark time on this change only. Not sure if this is really relevant.

File: src/geode/geometry/distance.cpp [hunk 9]
```diff
@@ -60,6 +60,27 @@ namespace
...
+    {
+        if( std::fabs(
+                point_to_v0_length + point_to_v1_length - segment_length )
+            < geode::global_epsilon )
+        {
+            return 0;
+        }
+        const auto p =
+            ( point_to_v0_length + point_to_v1_length + segment_length ) / 2;
+        auto area2 = p * ( p - point_to_v0_length ) * ( p - point_to_v1_length )
```
Comment (BotellaA) [inline src/geode/geometry/distance.cpp:76]:
```suggestion
        const auto area2 = p * ( p - point_to_v0_length ) * ( p - point_to_v1_length )
```

File: src/geode/geometry/intersection.cpp [hunk 10]
```diff
@@ -170,11 +168,9 @@ namespace geode
             line_plane_intersection( InfiniteLine3D{ segment }, plane );
         if( line_plane_result )
         {
-            double distance_to_segment;
-            std::tie( distance_to_segment, std::ignore ) =
-                point_segment_distance(
-                    line_plane_result.result.value(), segment );
-            if( distance_to_segment > global_epsilon )
+            if( new_point_segment_distance(
```
Comment (BotellaA) [inline src/geode/geometry/intersection.cpp:171]: Tu voulais pas aussi remplacer l'usage des coord barycentric dans les correctness?

File: src/geode/mesh/builder/solid_mesh_builder.cpp [hunk 11]
```diff
@@ -305,46 +277,65 @@ namespace
         const geode::PolyhedronVertex& polyhedron_vertex,
         geode::index_t new_vertex_id )
     {
-        const auto facets_vertices =
```
Comment (BotellaA) [inline src/geode/mesh/builder/solid_mesh_builder.cpp:308]: Why do you revert this change?

File: src/geode/mesh/builder/solid_mesh_builder.cpp [hunk 12]
```diff
@@ -815,6 +806,40 @@ namespace geode
         }
     }
 
+    template < index_t dimension >
+    void SolidMeshBuilder< dimension >::update_polyhedron_adjacencies(
```
Comment (BotellaA) [inline src/geode/mesh/builder/solid_mesh_builder.cpp:810]: Should we add this in surface too?

File: src/geode/geometry/intersection.cpp (hunk 10 above)
Comment (panquez) [reply src/geode/geometry/intersection.cpp:171]: Yes I will do it in a next PR

File: src/geode/mesh/builder/solid_mesh_builder.cpp (hunk 11 above)
Comment (panquez) [reply src/geode/mesh/builder/solid_mesh_builder.cpp:308]: Because the two versions are quasi similar in terms of Ir. I prefer the less verbose version.

File: src/geode/mesh/builder/solid_mesh_builder.cpp (hunk 12 above)
Comment (panquez) [reply src/geode/mesh/builder/solid_mesh_builder.cpp:810]: Yes good idea

**Review by BotellaA — APPROVED** [review]

File: src/geode/mesh/builder/solid_mesh_builder.cpp [hunk 13]
```diff
@@ -815,6 +806,40 @@ namespace geode
...
+                if( const auto adj = solid_mesh_.polyhedron_adjacent( id ) )
+                {
+                    const auto new_adjacent = old2new[adj.value()];
+                    if( new_adjacent == adj.value() )
+                    {
+                        continue;
+                    }
+                    if( new_adjacent == NO_ID )
+                    {
+                        unset_polyhedron_adjacent( id );
```
Comment (MelchiorSchuh) [inline src/geode/mesh/builder/solid_mesh_builder.cpp:832]: Are you sure that what you want to do isn't do_unset_polyhedron_adjacent ?
